docs/cross-platform.md

```diff
@@ -0,0 +1,119 @@
+# Cross-Platform Development with `esy`
+
+## Goal
+
+The goal for `esy` is to work equally well on Linux, Windows, and OSX - without needing special cases or hacks.
+
+The development workflow for `esy` should be the same for all of these platforms.
+
+In addition, because the development workflow is the same for all platforms, it means that setting up your projects to validate on CI is straightforward - the install/build/test process can look something like this:
+- `esy install`
+- `esy build`
+- `esy b jbuilder @runtest`
+- `esy release`
+
+## Challenges
+
+The primary challenge to making the OCaml ecosystem work and build on Windows is that build scripts rely heavily on 'bashisms', like `make`, bash shell scripts, etc.
+
+Those tools aren't available out-of-the-box on Windows. WSL is starting to offer more capabilities around this,
+but there isn't a seamless install story. An important piece of meeting our goal stated above is that the installation process
+for `esy` is simple - `npm install -g esy`.
+
+## Architecture
+
+To provide a seamless build experience for Windows, we use `cygwin` on provide a transparent layer for handling these 'bashisms'. You don't even have to know about it - we install it silently, keep it in a sandbox, and delegate commands to it under the hood - but you don't need to touch it for `esy`. As package authors improve their native Windows support, we may not even need this `cygwin` layer long-term, but it helps get us started with a seamless windows experience.
+
+The good news is that the majority of OCaml build tools do support building from a `cygwin` environment - and often, that is the recommended environment for building on Windows.
+
+Note that, even though our build commands on Windows are executed in a `cygwin` shell, the __binaries that are produced are truly native__. This applies both to all the packages we build, as well as the resultant binary for your project itself. Cygwin provides two compilers - a cygwin `gcc` compiler, and a `mingw` cross-compiler that compiles to native windows executables. We leverage the latter so the result is that `esy build` on Windows gives you native executables. This means that you can use `esy` to build apps natively for any platform!
```

Comment:
The way people often distinguish this from typical cygwin environments is a "mingw" toolchain. When people talk about cygwin then usually mean an environment that produces binaries that aren't pure windows binaries, but require cygwin1.dll to run.

Reply:
Good point - tweaked the wording here to try and make that more explicit.